src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -1816,48 +1812,85 @@ int D4NFilterObject::D4NFilterDeleteOp::delete_obj(const DoutPrefixProvider* dpp
   std::string head_oid_in_cache;
   rgw::d4n::CacheBlock block;
   if (!source->check_head_exists_in_cache_get_oid(dpp, head_oid_in_cache, attrs, block, y) && !block.deleteMarker) {
-    ldpp_dout(dpp, 0) << "D4NFilterObject::D4NFilterDeleteOp::" << __func__ << "(): calling next delete_obj" << dendl;
-    return next->delete_obj(dpp, y, flags);
+    ldpp_dout(dpp, 10) << "D4NFilterObject::" << __func__ << "(): head object not found; calling next->delete_obj" << dendl;
+    next->params = params;
+    int ret = next->delete_obj(dpp, y, flags);
+    ret = next->delete_obj(dpp, y, flags);
+    result = next->result;
+    return ret;
   } else {
     int ret = -1;
     bool objDirty = false;
     auto blockDir = source->driver->get_block_dir(); 
     std::string version, policy_prefix;
+    bool nullDelete = false;
```

Comment:
what if the object is dirty? Are these check common for dirty and clean objects?